### PR TITLE
chore: remove protoc grpc from generation config

### PR DESCRIPTION
**Pull request by JoeWang1127**
In this PR:
- Remove `protoc_version` and `grpc_version` from generation config.

### CODE REVIEW

**Review by blakeli0 — COMMENTED** [review]
I know we said that it's OK to not support running scripts directly. What is the effort to support it? Do we need to write additional code or just need to set `DOCKER_PROTOC_VERSION` and `DOCKER_GRPC_VERSION`?

**Comment by JoeWang1127** [comment]
> I know we said that it's OK to not support running scripts directly. What is the effort to support it? Do we need to write additional code or just need to set `DOCKER_PROTOC_VERSION` and `DOCKER_GRPC_VERSION`?

I think we need to download the the protoc/grpc and put them into a well-known location, just like the generator and formatter jar.

**Comment by diegomarquezp** [comment]
> * Remove instructions on running python script in local environment because the current setup will fail if `DOCKER_PROTOC_VERSION` or `DOCKER_GRPC_VERSION` is not set.

We don't write any additional code to support running the scripts directly.

**Comment by diegomarquezp** [comment]
> * Remove instructions on running python script in local environment because the current setup will fail if `DOCKER_PROTOC_VERSION` or `DOCKER_GRPC_VERSION` is not set.

@JoeWang1127 why not just explain the developer that these variables must be set?

**Comment by blakeli0** [comment]
> I think we need to download the the protoc/grpc and put them into a well-known location, just like the generator and formatter jar

Cool, if that's the case, maybe we can continue to support it by adding a few documentation to DEVELOPMENT.md? Feel free to not do it if you think there is not much use case for it anyway and it's not worth the effort.

**Comment by JoeWang1127** [comment]
> @JoeWang1127 why not just explain the developer that these variables must be set?

I don't think it's intuitive to set these environment variables only for running the script directly. 

The variable name is clearly indicating running in docker container, why setting these env if we're using python script?

**Comment by diegomarquezp** [comment]
> > @JoeWang1127 why not just explain the developer that these variables must be set?
> 
> I don't think it's intuitive to set these environment variables only for running the script directly.
> 
> The variable name is clearly indicating running in docker container, why setting these env if we're using python script?

I agree it's not intuitive. However, these variables are automatically set when building the image, so we must set them always, although manually for the case of local development. 

I remember we used the `DOCKER_` prefix was used to distinguish the variables from what's in the config file, but we are not using these config entries in the yaml since this PR. Maybe we can drop the prefix and rename them to `PROTOC_LOCATION`?

**Comment by JoeWang1127** [comment]
> Cool, if that's the case, maybe we can continue to support it by adding a few documentation to DEVELOPMENT.md? Feel free to not do it if you think there is not much use case for it anyway and it's not worth the effort.

I think we can create another pr to support the directly run. I feel it's too much for this PR.

> I remember we used the `DOCKER_ prefix` was used to distinguish the variables from what's in the config file, but we are not using these config entries in the yaml since this PR. Maybe we can drop the prefix and rename them to PROTOC_LOCATION? edit: this can be a follow up

SGTM.

**Comment by JoeWang1127** [comment]
I decided to keep the support of running python script directly in development guide.

File: .cloudbuild/library_generation/library_generation.Dockerfile [hunk 1]
```diff
@@ -88,24 +88,23 @@ WORKDIR /protoc
 RUN source /src/library_generation/utils/utilities.sh \
 	&& download_protoc "${PROTOC_VERSION}" "${OS_ARCHITECTURE}"
 # we indicate protoc is available in the container via env vars
-ENV DOCKER_PROTOC_LOCATION=/protoc
+ENV DOCKER_PROTOC_LOCATION=/protoc/protoc/bin
```
Comment (blakeli0) [inline .cloudbuild/library_generation/library_generation.Dockerfile:91]: Is duplicated `protoc` expected in the location?

Comment (JoeWang1127) [reply .cloudbuild/library_generation/library_generation.Dockerfile:91]: Yes.

Do you think we should reduce the duplicated directory?

Comment (blakeli0) [reply .cloudbuild/library_generation/library_generation.Dockerfile:91]: Yes, if it is not required, just `protoc/bin` is clearer IMO. I think we can unzip to the root folder in the [line below](https://github.com/googleapis/sdk-platform-java/pull/3390/files#diff-c6e48602f6abf1dd0e120f8ee5e22e146db761c31a44d02830b67ecf17f88309R108)?

Comment (JoeWang1127) [reply .cloudbuild/library_generation/library_generation.Dockerfile]: Done.

**Review by blakeli0 — APPROVED** [review]

**Pull request merged**